src/libraries/System.IO.Pipelines/ref/System.IO.Pipelines.cs

```diff
@@ -95,10 +95,12 @@ public static partial class StreamPipeExtensions
     public partial class StreamPipeReaderOptions
     {
         public StreamPipeReaderOptions(System.Buffers.MemoryPool<byte>? pool = null, int bufferSize = -1, int minimumReadSize = -1, bool leaveOpen = false) { }
+        public StreamPipeReaderOptions(bool waitForData, System.Buffers.MemoryPool<byte>? pool = null, int bufferSize = -1, int minimumReadSize = -1, bool leaveOpen = false) { }
```

Comment:
From .NET Framework design guidelines:

AVOID being inconsistent in the ordering of parameters in overloaded members. Parameters with the same name should appear in the same position in all overloads.
AVOID having two overloads of the same method that both use default parameters.
DO move all default parameters to the new, longer overload when adding optional parameters to an existing method

Reply:
OK I have no idea what I'm doing because adding it to the end didn't work.

Reply:
```C#
public class Foo
{
    public Foo(int x = 1, int y = 2) { }
    public Foo(int x = 1, int y = 2, int z) { } // This doesn't compile.
    public Foo(int x, int y, int z) { } // Is this the answer? No defaults?
}
```